main.sh

```diff
@@ -109,25 +125,42 @@ while read line; do
   fi
 done <<< "${usage}"
 
+# Allow long options like --this
+opts="${opts}-:"
+
```

Comment:
This is the trick. I saw it here: http://mywiki.wooledge.org/ComplexOptionParsing (second code block)